From 58ed121cea6a4eaef1ed41495d06bc6e1a4e2d08 Mon Sep 17 00:00:00 2001
From: Alex Avance <aravance@gmail.com>
Date: Sun, 3 Jun 2012 20:37:05 -0700
Subject: [PATCH 22/30] Add an option to change the device hostname (1/2).

This adds an option to modify the device hostname used
in ip resolution. This is useful when connecting to the
android device in a dynamic dhcp environment.

Change-Id: I44e4771f4415a32fe66867d4dd90f1ad138341e1
---
 .../core/java/com/android/server/ConnectivityService.java  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index 8c1f5d8e6e3..8ce0839e8b1 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -94,6 +94,7 @@ import android.os.ResultReceiver;
 import android.os.ServiceManager;
 import android.os.ServiceSpecificException;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.provider.Settings;
@@ -171,6 +172,8 @@ import java.util.Objects;
 import java.util.SortedSet;
 import java.util.TreeSet;
 
+import lineageos.providers.LineageSettings;
+
 /**
  * @hide
  */
@@ -856,6 +859,10 @@ public class ConnectivityService extends IConnectivityManager.Stub
         mMultinetworkPolicyTracker = createMultinetworkPolicyTracker(
                 mContext, mHandler, () -> rematchForAvoidBadWifiUpdate());
         mMultinetworkPolicyTracker.start();
+
+        String hostname = LineageSettings.Secure.getString(context.getContentResolver(),
+                LineageSettings.Secure.DEVICE_HOSTNAME);
+        SystemProperties.set("net.hostname", hostname);
     }
 
     private Tethering makeTethering() {
-- 
2.17.1

